### PR TITLE
some vault fixes

### PR DESCRIPTION
**Pull request by RichiCoder1**
~I noted this extra write, but I'm not sure this will actually fix the issue. I couldn't reproduce the pub key issue via e2e.~

I updated the tests to use the same `ECDSAVerifier` that cosign uses and sure enough it started failing. I ended up shifting the hash to hash locally instead of using Vault's hash endpoint and it seems happy now.

### CODE REVIEW

File: test/e2e/kms/vault_test.go [hunk 1]
```diff
@@ -104,6 +106,40 @@ func (suite *VaultSuite) TestSign() {
...
+	ok = ecdsa.VerifyASN1(pubKey, hash[:], sig)
+	assert.True(suite.T(), ok)
+
+	ok = ecdsa.VerifyASN1(key, hash[:], sig)
+	assert.True(suite.T(), ok)
+
+	verifier := signature.ECDSAVerifier{
+		Key:     pubKey,
+		HashAlg: crypto.SHA256,
+	}
```
Comment (RichiCoder1) [inline test/e2e/kms/vault_test.go:139]: In theory this is exactly what is happening in `cosign verify`. I wonder if something is getting lost in  translation.

**Comment by RichiCoder1** [comment]
Unrelated: It looks like `pkg/oauthflow/flow_test.go` is experiencing a race condition

**Review by dlorenc — APPROVED** [review]
Wow, thanks for the debugging!

**Comment by dlorenc** [comment]
> Unrelated: It looks like `pkg/oauthflow/flow_test.go` is experiencing a race condition

Opened an issue.

**Pull request merged**